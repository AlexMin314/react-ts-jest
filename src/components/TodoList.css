.TodoList {
    text-align: left;
}

.TodoList--Item--Done {
    text-decoration: line-through;
}