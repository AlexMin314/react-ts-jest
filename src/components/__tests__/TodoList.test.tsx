import * as React from 'react';
import {TodoList} from '../TodoList';
import {TodoStore} from '../../stores/TodoStore';
import * as renderer from 'react-test-renderer';


// jest.mock('../../stores/TodoStore');

describe('TodoList', () => {
    it('should contain added todo item', () => {
        // Given
        const todoStore = new TodoStore();
        const component = renderer.create(<TodoList todoStore={todoStore}/>);

        // When
        todoStore.addTodo('hello');

        // Then
        expect(component.toJSON()).toMatchSnapshot();
    });
});
