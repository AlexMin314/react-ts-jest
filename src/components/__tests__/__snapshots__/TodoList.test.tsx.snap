// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList should contain added todo item 1`] = `
<ul
  className="TodoList"
>
  <li
    className=""
  >
    <input
      checked={false}
      onClick={[Function]}
      type="checkbox"
    />
    hello
  </li>
</ul>
`;
